src/main/java/org/traccar/broadcast/RedisBroadcastService.java

```diff
@@ -0,0 +1,206 @@
+        if (message.getDevice() != null) {
+            listeners.forEach(listener -> listener.updateDevice(false, message.getDevice()));
+        } else if (message.getPosition() != null) {
+            listeners.forEach(listener -> listener.updatePosition(false, message.getPosition()));
+        } else if (message.getUserId() != null && message.getEvent() != null) {
+            listeners.forEach(listener -> listener.updateEvent(false, message.getUserId(), message.getEvent()));
+        } else if (message.getCommandDeviceId() != null) {
+            listeners.forEach(listener -> listener.updateCommand(false, message.getCommandDeviceId()));
+        } else if (message.getChanges() != null) {
+            var iterator = message.getChanges().entrySet().iterator();
+            if (iterator.hasNext()) {
+                var first = iterator.next();
+                if (iterator.hasNext()) {
+                    var second = iterator.next();
+                    listeners.forEach(listener -> listener.invalidatePermission(
+                            false,
+                            Permission.getKeyClass(first.getKey()), first.getValue(),
+                            Permission.getKeyClass(second.getKey()), second.getValue()));
+                } else {
+                    listeners.forEach(listener -> listener.invalidateObject(
+                            false,
+                            Permission.getKeyClass(first.getKey()), first.getValue()));
+                }
+            }
+        }
+    }
+
+    @Override
+    public void start() throws IOException {
+        service.submit(receiver);
+    }
+
+    @Override
+    public void stop() {
+        service.shutdown();
+    }
+
+    private final Runnable receiver = new Runnable() {
+        @Override
+        public void run() {
+            subscriberJedis.subscribe(new JedisPubSub() {
+                @Override
+                public void onMessage(String channel, String message) {
+                    try {
+                        String[] parts = message.split(":", 2);
+                        if (channel == pubsubChannel && parts.length == 2 && !id.equals(parts[0])) {
+                            handleMessage(objectMapper.readValue(parts[1], BroadcastMessage.class));
+                        }
+                    } catch (IOException e) {
+                        LOGGER.warn("Broadcast handleMessage failed", e);
+                    }
+                }
+            }, pubsubChannel);
+
+            while (!service.isShutdown()) {
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    break;
+                }
+            }
+
+            try {
+                subscriberJedis.close();
+                publisherJedis.close();
+            } catch (Exception e) {
+                LOGGER.warn("Failed to close pubsub", e);
+                throw new RuntimeException(e);
+            }
```

Comment:
It's not a good idea to catch generic `Exception`. And you should either throw or log. Otherwise you will have duplication in logs. I prefer throw because that's what multicast does.